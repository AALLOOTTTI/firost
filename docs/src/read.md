---
title: await read(filepath)
---

Returns the textual content of the file located at `filepath`.

## Examples

```js
const content = await read('./src/css/style.css');
```
