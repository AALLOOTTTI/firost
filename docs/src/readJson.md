---
title: readJson
---

<div class="lead">
  Read the content of a local <code>.json</code> file and return it as a JavaScript
  object.
</div>

`await readJson(filepath)`

## Examples

```js
const data = await readJson('./records.json');
```

## Notes

Note that it will read the whole content before returning it, so might not be
suitable for large files (in which case you might want to use streams instead).
