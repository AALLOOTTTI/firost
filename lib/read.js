const _ = require('golgoth/lib/_');
const fs = require('fs-extra');
const absolute = require('./absolute');

/**
 * Read any file on disk
 * @param {string} userFilepath Filepath of the file to read
 * @returns {string} Content of the file read
 **/
module.exports = async function read(userFilepath) {
  const filepath = absolute(userFilepath);
  const content = await fs.readFile(filepath);
  return _.trim(content.toString('utf8'));
};
