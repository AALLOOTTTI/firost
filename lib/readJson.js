import read from './read';
import error from './error';

/**
 * Read a JSON file on disk and return its parsed content.
 * @param {String} source Path to the JSON file
 * @return {Promise.<Object>} The parsed content of the JSON file
 **/
const module = async function readJson(source) {
  const content = await read(source);
  try {
    return JSON.parse(content);
  } catch (err) {
    throw error('ERROR_NOT_JSON', `${source} is not a valid JSON file`);
  }
};

export default module;
