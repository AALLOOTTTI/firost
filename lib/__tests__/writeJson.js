import module from '../writeJson';
import helper from '../test-helper';
import readJson from '../readJson';
import read from '../read';
const tmpPath = helper.tmpPath('writeJson');

describe('writeJson', () => {
  beforeEach(async () => {
    await helper.clearTmpDirectory('writeJson');
  });
  it('should create a file with the given content', async () => {
    const destination = `${tmpPath}/destination.json`;
    await module({ foo: 'bar' }, destination);

    const actual = await readJson(destination);
    expect(actual).toHaveProperty('foo', 'bar');
  });
  it('should overwrite an existing file', async () => {
    const destination = `${tmpPath}/destination.json`;
    // Set the first content
    await module({ foo: 'bar' }, destination);

    await module({ foo: 'baz' }, destination);
    const actual = await readJson(destination);
    expect(actual).toHaveProperty('foo', 'baz');
  });
  it('should create nested directories', async () => {
    const destination = `${tmpPath}/foo/bar/destination.json`;
    await module({ foo: 'bar' }, destination);

    const actual = await readJson(destination);
    expect(actual).toHaveProperty('foo', 'bar');
  });
  it('should pretty print the output', async () => {
    const destination = `${tmpPath}/destination.json`;
    await module({ foo: 'bar', bar: 'baz' }, destination);

    const actual = await read(destination);
    expect(actual).toMatchSnapshot();
  });
  it('should sort the object keys, to ease diffing', async () => {
    const destination = `${tmpPath}/destination.json`;
    await module(
      { zoo: 'bar', coo: 'bar', moo: 'bar', foo: 'bar' },
      destination
    );

    const actual = await read(destination);
    expect(actual).toMatchSnapshot();
  });
  it('should not sort the object keys with { sort: false }', async () => {
    const destination = `${tmpPath}/destination.json`;
    await module(
      { zoo: 'bar', coo: 'bar', moo: 'bar', foo: 'bar' },
      destination,
      { sort: false }
    );

    const actual = await read(destination);
    expect(actual).toMatchSnapshot();
  });
});
