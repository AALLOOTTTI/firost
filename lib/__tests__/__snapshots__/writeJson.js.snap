// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeJson should not sort the object keys with { sort: false } 1`] = `
"{
  \\"zoo\\": \\"bar\\",
  \\"coo\\": \\"bar\\",
  \\"moo\\": \\"bar\\",
  \\"foo\\": \\"bar\\"
}"
`;

exports[`writeJson should pretty print the output 1`] = `
"{
  \\"bar\\": \\"baz\\",
  \\"foo\\": \\"bar\\"
}"
`;

exports[`writeJson should sort the object keys, to ease diffing 1`] = `
"{
  \\"coo\\": \\"bar\\",
  \\"foo\\": \\"bar\\",
  \\"moo\\": \\"bar\\",
  \\"zoo\\": \\"bar\\"
}"
`;
